front/components/assistant_builder/NamingScreen.tsx

```diff
@@ -168,35 +177,83 @@ export default function NamingScreen({
   ]);
 
   // Description suggestions handling
-  const [descriptionSuggestions, setDescriptionSuggestions] =
+  /*const [descriptionSuggestions, setDescriptionSuggestions] =
     useState<BuilderSuggestionsType>({
       status: "unavailable",
       reason: "irrelevant",
     });
 
   const [descriptionSuggestionsIndex, setDescriptionSuggestionIndex] =
-    useState(0);
+    useState(0);*/
 
-  const updateDescriptionSuggestions = useCallback(async () => {
+  const [generatingDescription, setGeneratingDescription] = useState(false);
+  const [descriptionIsGenerated, setDescriptionIsGenerated] = useState(false);
+
+  const suggestDescription = useCallback(async () => {
+    setGeneratingDescription(true);
     const descriptionSuggestions = await getDescriptionSuggestions({
       owner,
       instructions: builderState.instructions || "",
       name: builderState.handle || "",
     });
     if (descriptionSuggestions.isOk()) {
-      setDescriptionSuggestions(descriptionSuggestions.value);
+      const suggestion =
+        descriptionSuggestions.value.status === "ok" &&
```

Comment:
Good point. This is debatable, since there are cases where user input is correct, the API call works, the dust app is run, and the LLM decides it's not time to create suggestions (e.g. because of the content of the instructions). So it's not exactly an error, the API call worked and is saying "ok, I looked, with this data it's not appropriate to give you suggestions" (but the request body was correct, with valid arguments, and no way to tell beforehand)

WDYT?

Reply:
From a FE standpoint, we don't need to know all this conditional logic around the LLM deciding whether or not it's time to generate a description.

I'd suggest to go with an empty array in those scenarios.

Reply:
Well, depending on the reason for the empty array, the front end will need to react differently. Agree that's debatable, boils down to deciding what are valid answers from the endpoint and what are "errors" and I agree the duplicate status is indeed weird. But changing API scope is not for this PR. We can discuss it further for another PR if you beleive it's warranted